torch/library.py

```diff
@@ -827,6 +829,54 @@ def register_kernel(
     return _impl(op, device_types, func, lib=lib, disable_dynamo=True)
 
 
+def register_autocast(
+    op: _op_identifier,
+    device_type: str,
+    cast_inputs: Optional[_dtype] = None,
+    /,
+    *,
+    lib: Optional[Library] = None,
+):
+    if not isinstance(
+        op, (str, torch._ops.OpOverload, torch._library.custom_ops.CustomOpDef)
+    ):
+        raise ValueError(
+            f"register_autocast(op): got unexpected type for op: {type(op)}"
+        )
+    if isinstance(op, torch._ops.OpOverload):
+        op = op._name
+    opdef = _maybe_get_opdef(op)
+    if opdef is not None:
+        return opdef.register_autocast(device_type, cast_inputs)
+
+    if cast_inputs is not None:
+        assert isinstance(op, str)
+        qualname = op
+        _op = torch._library.utils.lookup_op(qualname)
+
+        namespace, opname = torch._library.utils.parse_namespace(qualname)
+        if lib is None:
+            lib = Library(namespace, "FRAGMENT")
+            _keep_alive.append(lib)
+
+        def kernel(_, *args, **kwargs):
+            nonlocal _op
```

Comment:
I don't think you need nonlocal here? I thought that was only needed if you mutate _op